src/actions/debuggee.js

```diff
@@ -5,10 +5,19 @@
 // @flow
 
 import type { Action, ThunkArgs } from "./types";
+import { closeTabsForMissingThreads } from "./tabs";
+import { getMainThread } from "../reducers/pause";
 
 export function updateWorkers() {
-  return async function({ dispatch, client }: ThunkArgs) {
+  return async function({ dispatch, getState, client }: ThunkArgs) {
     const { workers } = await client.fetchWorkers();
     dispatch(({ type: "SET_WORKERS", workers }: Action));
+
+    closeTabsForMissingThreads(
+      dispatch,
```

Comment:
Sure, I definitely want to improve this but I'm not sure how --- I couldn't find much documentation online or examples in debugger.html of how to either compose actions that are in separate components, or to trigger actions transitively in response to another change to the redux store.  Is there a way for closeTabsForMissingThreads() to have the same signature of other action functions?  Having it return an array of actions instead of doing the dispatching itself would still leave it with different usage from the usual action function.

Reply:
How costly is `await client.fetchWorkers()` we can either pass it or re-fetch it in the `closeTabsForMissingThreads` action.
I can try to change the signature :)